lifemonitor/api.yaml

```diff
@@ -119,7 +211,129 @@ paths:
         '204':
           description: Deleted
         '400':
-          description: Invalid input
+          $ref: '#/components/responses/BadRequest'
+
+  /suites/{suite_uuid}/status:
+    get:
+      summary: >
+        Get status of latest build for this test suite for each known instance
+      parameters:
+        - $ref: '#/components/parameters/suite_uuid'
+      responses:
+        '200':
+          description: TestSuiteStatus
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TestSuiteStatus'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+  /suites/{suite_uuid}/instances:
```

Comment:
For external testing, I guess we want to create these objects from the user-provided test metadata (paying attention to duplicates). For internal testing we need a `SuiteInstance` that models the internal test instance. When should this be created?

Reply:
Perhaps we can call "internal" test instances "managed" and "external" test instances "unmanaged", from the perspective of LifeMonitor?  "Hosted" might be another alternative.

Assumin the terminology above, an unmanaged test instance could be created:
1. from the metadata in a workflow-ro-crate;
2. through a POST to `/suites/{suite_uuid}/instances`, which at the moment isn't in the spec.

On the other hand, a managed test instance has to be created by LifeMonitor itself in response to the workflow owner's request.  I haven't thought much regarding how this should be requested through the API, but it could be reasonable to expect a POST to the same resource (`/suites/{suite_uuid}/instances`) but with different parameters (e.g. `instance_type: "managed").

Reply:
Sorry, I got a bit confused between instance and service. In my previous comment I was actually thinking about services, rather than instances. I thought that the absence of a POST here was a feature, since:

1. That would allow to create a new service via the `service` property of `SuiteInstance`. However, in the unmanaged case, the service is already defined in the test metadata (so the two definitions would clash), while in the managed case it's supposed to be handled internally by LM.

2. Until now we've assumed that a test execution would be triggered by a new workflow or a workflow update.

Now I see that a POST for a test instance can be a way to allow the user to trigger a test execution at any time, but users should not be allowed to create services this way. So if we do add a POST here we should probably mark the `service` property as `readOnly` in `SuiteInstance`.

Reply:
I posted a new comment but then I realized I got confused with the terminology, so I deleted it. Apologies if you got an email for that from GitHub, just ignore it :-)